### DISCUSSION TITLE
Fetch and display Page Titles dynamically

### DISCUSSION THREAD

**Discussion by kmanojkumar** [outline/outline#2744] [Ideas]
Hi @tommoor, what do you think about displaying the Page's Title when an http link is pasted in Editor, instead of just displaying the link as it is (current behaviour).  

**Benefit:** This will provide readers some useful insights about the Page in a friendly-way. Like pasting GitHub/GitLab issue links displays Issue Title, Issue ID, Project Name by fetching the Page's Title.

Please refer the example below:
![image](https://user-images.githubusercontent.com/14366356/140658459-0e5fd9fd-a99c-4dda-8076-36f5ba2630fb.png)

Additionally, on hover a link preview of Page Title + Page Description + Meta Image may be shown.

Extends #1884.

**Comment by tommoor** [comment]
It's a good idea, as is often the case the question comes down to ux. How should this be presented? What is displayed while the title is loading? Etc 😀

**Comment by dkkb** [comment]
I have the same idea, what about this? @tommoor 

![image](https://user-images.githubusercontent.com/82504881/141737938-3eb7d1b5-74cb-4491-a4ef-d6ff875f3be0.png)

**Comment by chb-ioki** [comment]
Hello everyone,

I would also be interested in this topic and I am also eager to help.

For me, it would be interesting for GitHub and GitLab ticket URLs.

I personally would prefer a more minimalistic solution like turning

[ https://github.com/outline/outline/issues/1](https://github.com/outline/outline/issues/1)

into

**[Can't use multiple Slack teams right now (outline/outline #1) (closed)](https://github.com/outline/outline/issues/1)**

Keeping it simple has multiple advantages imho:

- It will not interfere with general formatting of the document
- It can lazy load and during loading and in case of error one can just keep the original URL (aka graceful fallback)

What do you think?

Cheers,
Christian

**Reply by dkkb** [reply]
I think that in terms of importance, the sorting should be as follows:

1. Link URL
2. Title of the link URL
3. Summary of the content of the link URL
4. First Image within the link content

The link URL is highly important, as often we need to first determine if the source of the link is from a trusted website.

**Reply by chb-ioki** [reply]
I think we also have two different things in mind:

I understand it in that way, that you would like to show opengraph/twitter card-like meta data for a link as a block, just like what happens when you post a link on a social feed like facebook or twitter.

This is also a nice idea!

My personal feature request was something else though.
In this case I wanted to replace a URL in a text with the title of site that is linked without rendering an extra box with meta information (images, etc.). You can find a feature like this in GitHub or GitLab markdown when you link to an issue or a PR.

So these are two slightly different features. Which is not a bad thing - but I think they could be handled individually.
